crates/runtime/src/lib.rs

```diff
@@ -377,48 +396,84 @@ impl Runtime {
         self.register_metrics_table(self.prometheus_registry.is_some())
             .await?;
 
+        // Start Http server
+        let cloned_tls_config = tls_config.clone();
+        let cloned_config = config.clone();
         let http_auth = endpoint_auth.http_auth.clone();
-        let http_server_future = tokio::spawn(http::start(
-            config.http_bind_address,
-            Arc::clone(&self),
-            config.clone().into(),
-            tls_config.clone(),
-            http_auth,
-        ));
-
-        // Spawn the metrics server in the background
+        let self_ref = Arc::clone(&self);
+
+        let http_future = self
```

Comment:
Axum supports `with_graceful_shutdown` that should be added similar to Tonic servers (Flight, OpenTelemetry). Will add next